src/core/thread/mle.hpp

```diff
@@ -662,6 +662,12 @@ class Mle : public InstanceLocator, private NonCopyable
      */
     void RemoveDelayedDataResponseMessage(void);
 
+    /**
+     * This method removes MLE Data Request from Delayed Message Queue for a given destination.
+     *
+     */
+    void RemoveDelayedDataRequestMessage(const Ip6::Address &aDestination);
```

Comment:
How about defining in the `protected` section (or maybe even `private`?). I don't think anyone other than `Mle` would need to use this method?